### PR TITLE
doc: release-notes-3.6: add release note about -Wdouble-promotion

### PR DESCRIPTION
**Pull request by XenuIsWatching**
This adds a release note about -Wdouble-promotion now being a default warning flag.

### CODE REVIEW

**Review by MaureenHelm — DISMISSED** [review]
A lot of work went into this, worth a mention in #68994

File: doc/releases/release-notes-3.6.rst [hunk 1]
```diff
@@ -290,6 +290,9 @@ Build system and infrastructure
 
 * Deprecated ``CONF_FILE`` ``prj_<build>.conf`` build type.
 
+* Added `-Wdouble-promotion` as a default warning when compiling to warn developers with
+  single-percision floats easily being promoted to double-percision.
```
Comment (henrikbrixandersen) [inline doc/releases/release-notes-3.6.rst:294]:
```suggestion
  single-percision floats easily being promoted to double-precision.
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-3.6.rst (hunk 1 above)
Comment (XenuIsWatching) [reply doc/releases/release-notes-3.6.rst:294]: i r no gud spellar :/

**Review by henrikbrixandersen — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
Good catch @henrikbrixandersen

**Pull request merged**